src/setup/init.cpp

```diff
@@ -475,3 +480,40 @@ void incflo::InitialProjection()
         PrintMaxValues(time);
     }
 }
+
+void incflo::setup_level_mask()
```

Comment:
You don't have to implement your own.  You can call

```
    iMultiFab makeFineMask (const BoxArray& cba, const DistributionMapping& cdm,
                            const BoxArray& fba, const IntVect& ratio,
                            int crse_value = 0, int fine_value = 1);
```

in AMReX_MultiFabUtil.H.  Note that `iMultiFab` has a move constructor that allows you to store the returned iMultiFab in a unique_ptr.

Reply:
Thanks Weiqun! I'm trying to use the makeFineMask but I can't figure it out. This chunk of code doesn't work...how do you suggest doing this?

``` std::unique_ptr<iMultiFab> imask;
      *imask = makeFineMask(grids[lev], dmap[lev], grids[lev+1], IntVect(2), 0, 1);
      level_mask[lev] = std::move(imask); //level_mask[lev].reset(imask)```

where `Vector<std::unique_ptr<iMultiFab>> level_mask;`

Reply:
I think something like this should work.
```
level_mask[lev].reset(new iMultiFab(makeFineMask(..)));
```